qiskit/transpiler/passes/routing/lookahead_swap.py

```diff
@@ -303,14 +302,12 @@ def _copy_circuit_metadata(source_dag, coupling_map):
     Generate only a single qreg in the output DAG, matching the size of the
     coupling_map.
     """
-    target_dag = DAGCircuit()
-    target_dag.name = source_dag.name
+    target_dag = source_dag._copy_circuit_metadata()
 
-    for creg in source_dag.cregs.values():
-        target_dag.add_creg(creg)
-
-    device_qreg = QuantumRegister(len(coupling_map.physical_qubits), 'q')
-    target_dag.add_qreg(device_qreg)
+    len_not_allocated_qubits = len(coupling_map.physical_qubits) - len(target_dag.qubits)
+    if len_not_allocated_qubits:
+        device_qreg = QuantumRegister(len_not_allocated_qubits, 'q')
+        target_dag.add_qreg(device_qreg)
```

Comment:
Is this still needed (this might have been from before `ExtendWithAncilla`)? If so, can we move it outside this function (maybe to L128)? It'd be unusual to have a `LookaheadSwap._copy_circuit_metadata` that's almost but not exactly the same as `DAGCircuit._copy_circuit_metadata`.

Reply:
not sure.. I just wanted to preserve the semantic from before. Let me see if testing fails without the extension.

Reply:
You are right, I think. The mappers work under the assumption that the circuit was already enlarged. The full function can be removed, 1fb795914